File: classfields/public-api-tests/src/test/java/ru/auto/tests/publicapi/carfax/DeleteCommentTest.java
package ru.auto.tests.publicapi.carfax;

import com.carlosbecker.guice.GuiceModules;
import com.carlosbecker.guice.GuiceTestRunner;
import com.google.inject.Inject;
import io.qameta.allure.Owner;
import io.qameta.allure.junit4.DisplayName;
import org.junit.Rule;
import org.junit.Test;
import org.junit.rules.RuleChain;
import org.junit.runner.RunWith;
import ru.auto.tests.passport.account.Account;
import ru.auto.tests.passport.manager.AccountManager;
import ru.auto.tests.publicapi.ApiClient;
import ru.auto.tests.publicapi.adaptor.PublicApiAdaptor;
import ru.auto.tests.publicapi.module.PublicApiModule;

import static ru.auto.tests.commons.util.Utils.getRandomString;
import static ru.auto.tests.publicapi.ResponseSpecBuilders.validatedWith;
import static ru.auto.tests.publicapi.api.CarfaxApi.DeleteCommentCarfaxOper.BLOCK_ID_PATH;
import static ru.auto.tests.publicapi.api.CarfaxApi.DeleteCommentCarfaxOper.VIN_PATH;
import static ru.auto.tests.publicapi.consts.Owners.TIMONDL;
import static ru.auto.tests.publicapi.model.AutoApiOffer.CategoryEnum.CARS;
import static ru.auto.tests.publicapi.ra.RequestSpecBuilders.defaultSpec;
import static ru.auto.tests.publicapi.ra.ResponseSpecBuilders.shouldBeEmptyJson;

@DisplayName("DELETE /carfax/user/comment")
@RunWith(GuiceTestRunner.class)
@GuiceModules(PublicApiModule.class)
public class DeleteCommentTest {

    private static final String VIN = "X9FLXXEEBLES67719";

    @Rule
    @Inject
    public RuleChain defaultRules;

    @Inject
    private AccountManager accountManager;

    @Inject
    private PublicApiAdaptor adaptor;

    @Inject
    private ApiClient api;

    @Test
    @Owner(TIMONDL)
    public void shouldSee200DeleteComment() {
        Account account = accountManager.create();
        String sessionId = adaptor.login(account).getSession().getId();
        String blockId = getRandomString();
        String offerId = RawReportUtils.createTestOffer(adaptor, account, sessionId);
        adaptor.waitUserOffersActive(sessionId, CARS, VIN);

        adaptor.createCarfaxComment(sessionId, VIN, blockId, getRandomString(100));

        api.carfax().deleteCommentCarfax().reqSpec(defaultSpec())
                .reqSpec(requestSpecBuilder -> {
                    requestSpecBuilder.addQueryParam(VIN_PATH, VIN);
                    requestSpecBuilder.addQueryParam(BLOCK_ID_PATH, blockId);
                })
                .xSessionIdHeader(sessionId)
                .execute(validatedWith(shouldBeEmptyJson()));
    }
}

File: classfields/public-api-tests/src/test/java/ru/auto/tests/publicapi/comments/DeleteCommentTest.scala
package ru.auto.tests.publicapi.comments

import com.carlosbecker.guice.{GuiceModules, GuiceTestRunner}
import com.google.inject.Inject
import io.qameta.allure.Owner
import io.qameta.allure.junit4.DisplayName
import org.junit.rules.RuleChain
import org.junit.runner.RunWith
import org.junit.{Rule, Test}
import ru.auto.tests.commons.restassured.ResponseSpecBuilders.shouldBe200OkJSON
import ru.auto.tests.passport.manager.AccountManager
import ru.auto.tests.publicapi.ApiClient
import ru.auto.tests.publicapi.ResponseSpecBuilders.validatedWith
import ru.auto.tests.publicapi.adaptor.PublicApiAdaptor
import ru.auto.tests.publicapi.consts.Owners.TIMONDL
import ru.auto.tests.publicapi.module.PublicApiModule
import ru.auto.tests.publicapi.ra.RequestSpecBuilders.defaultSpec

import scala.annotation.meta.getter

@DisplayName("DELETE /comments/{topicGroup}/{topicId}/{commentId}")
@GuiceModules(Array(classOf[PublicApiModule]))
@RunWith(classOf[GuiceTestRunner])
class DeleteCommentTest {

  private val TOPIC_GROUP = "review"
  private val TOPIC_ID = "8498572144589540098"

  @(Rule @getter)
  @Inject
  val defaultRules: RuleChain = null

  @Inject
  private val api: ApiClient = null

  @Inject
  private val accountManager: AccountManager = null

  @Inject
  private val adaptor: PublicApiAdaptor = null

  @Test
  @Owner(TIMONDL)
  def shouldDeleteComment(): Unit = {
    val account = accountManager.create()
    val sessionId = adaptor.login(account).getSession.getId
    val addResponse = adaptor.createComment(sessionId, TOPIC_GROUP, TOPIC_ID)

    api.comments.deleteComment().reqSpec(defaultSpec)
      .topicGroupPath(TOPIC_GROUP)
      .topicIdPath(TOPIC_ID)
      .xSessionIdHeader(sessionId)
      .commentIdPath(addResponse.getComment.getId)
      .execute(validatedWith(shouldBe200OkJSON))
  }
}
